File: src/main/java/ar/com/flow/bankaccount/withdrawal/CurrentFundsLimit.java
package ar.com.flow.bankaccount.withdrawal;

import ar.com.flow.bankaccount.BankAccount;
import ar.com.flow.money.Money;

public class CurrentFundsLimit implements WithdrawalLimit {
    public boolean accepts(Withdrawal withdrawal) {
        return withdrawal.previewBalanceAfter().isGreaterThanOrEqual(withdrawal.getAmount());
    }

    @Override
    public boolean accepts(BankAccount account, Money amount) {
        var previewBalanceAfter = account.getBalance().minus(amount);

        return previewBalanceAfter.isGreaterThanOrEqual(amount);
    }
}

File: src/main/kotlin/ar/com/flow/bankaccount/withdrawal/CurrentFundsLimit.kt
package ar.com.flow.bankaccount.withdrawal

import ar.com.flow.bankaccount.BankAccount
import ar.com.flow.money.Money

class CurrentFundsLimit : WithdrawalLimit {
    override fun accepts(account: BankAccount, amount: Money): Boolean {
        val previewBalanceAfter = account.balance.minus(amount)
        return previewBalanceAfter.isGreaterThanOrEqual(amount)
    }
}